### ISSUE TITLE
when cmor var is not in db cmor table is assigned incorrectly

### ISSUE THREAD

**Issue by paolap** [ACCESS-Community-Hub/ACCESS-MOPPeR#120]
Accidentally I discovered this bug when trying to pass mappings for newly defineded cmor var before adding the new definitions to the database.
So ideally this shouldn't occur frequently, however, it should be fixed as instead of the cmor table these records present the standard_name.
So clearly something is wrong!!

**Comment by paolap** [comment]
I think the issue is with variables coming out of the varilist command already mapped to a variable, for example:

fld_s08i223;mrsol;kg m-2;time depth lat lon;mon;atmos;area: time: mean;CMIP6_Emon;float32;663552;12;cw323a.pm;SOIL MOISTURE CONTENT IN A LAYER;mass_content_of_water_in_soil_layer

There are 3 variables in db which match this input var, version, frequency
mrsol, mrso, mrsos
the difference is that the second and third one have a calculation applied to them and with the latest changes we don't exclude these anymore in parse_vars().
So I need to adapt the code to be consistent and potentially remove the first mapping attempt in varlist.

**Comment by paolap** [comment]
There are two points where this fails
in add_var()
if row[1] == '' :
        db_log.debug(f"Assign cmor_var: {match}")
        row[1] = match[0]
These lines assign the matched cmor var name only if not already present which means that in the example above mrsol never gets replaced by mrso and mrsos.

The other problem that even fixing this we're not assigning the calculation, as before we were working out calculation separately, so I need two changes
1) return calculation always in sql query
2) add calculation in add_row

**Comment by paolap** [comment]
While this is progressing I'm finding two major issues with the all logic

1) I'm reading the earliest file in rows and then I modify the same list of lists.
This is a problem as when I try to use the same "rows" for derived variables the code breaks, as I now swapped positions of some items and added others. I want to use the original values to make sure I'm not introducing some "bias"
2) using lists is also problematic as anytime I change the order of the fields I need to make sure I propagate the changes across the all code. Maybe using dictionaries would help then it would be impossible to retrieve the wrong field

**Comment by paolap** [comment]
Moved to dictionaries from list, it really simplified a lot of logic as now I can treat most results returned from matching in the same way. It made also more compact the code to actually write results to file. Finally it's easier to understand what code is trying to do.
I also added a check (currently in check_realm which might become redundant) if a variable definition is not present in the db cmor table it now issues a warning and leave the cmor_table empty. 

This should happen rarely to a generic users unless they're explicitly adding new variables and forget to load them to the db.
